dataprofiler/profilers/profile_builder.py

```diff
@@ -660,7 +664,39 @@ def _remove_data_labelers(self):
         :return: data_labeler used for unstructured labelling
         :rtype: DataLabeler
         """
-        raise NotImplementedError()
+        data_labeler = None
+        data_labeler_options = None
+
+        # determine if the data labeler is enabled
+        use_data_labeler = True
+        if self.options and isinstance(self.options, (StructuredOptions,
+                                                      UnstructuredOptions)):
+            data_labeler_options = self.options.data_labeler
+            use_data_labeler = data_labeler_options.is_enabled
+
+        # remove the data labeler from options
+        if use_data_labeler and data_labeler_options is not None \
+                and data_labeler_options.data_labeler_object is not None:
+            data_labeler = data_labeler_options.data_labeler_object
+            data_labeler_options.data_labeler_object = None
+
+        # get all profiles, unstructured is a single profile and hence needs to
+        # be in a list, whereas structured is a dict and needs to be a list
+        profilers = [self._profile]
+        if isinstance(self, StructuredProfiler):
+            profilers = self._profile.values()
+
+        # Remove data labelers for all columns
+        for profiler in profilers:
+            if isinstance(self, StructuredProfiler):
+                profiler = profiler.profiles['data_label_profile']
```

Comment:
What is this doing? This could use a comment its a bit confusing?

Reply:
added comment